### DISCUSSION TITLE
Item timing

### DISCUSSION THREAD

**Discussion by umitgoren** [ChebanovDD/MatchSweets#6] [Q&A] [answered]
Hi there,

When I change the value of `MoveDuration` in MoveJob.cs to a  large value such as `1f,` I can notice that diagonal falling items overlap with each other while falling down. How can I synch them, so that they can fall down without overlap, one after the other? Do you think that the speed should be calculated dynamically based on the total distance that each item covers? Thanks.

**Comment by ChebanovDD** [comment, marked as answer]
Hi,

I think there are two ways to prevent overlaps.

## First way
The first way is to change the `PrependInterval` based on the item `RowIndex` (for example).

```csharp
public class ItemsMoveJob : MoveJob
{
    ...

    public override async UniTask ExecuteAsync(CancellationToken cancellationToken = default)
    {
        var itemsSequence = DOTween.Sequence();

        foreach (var itemData in _itemsData)
        {
            var itemMoveTween = CreateItemMoveTween(itemData);

            // Use your own math here.
            var rowIndexBasedInterval = (1 - itemData.RowIndex / 10.0f) * IntervalDuration;
            
            _ = itemsSequence
                .Join(itemMoveTween)
                .PrependInterval(itemMoveTween.Duration() * rowIndexBasedInterval);
        }

        ...
    }
}
```

> Maybe you will need to calculate the speed based on the total distance that each item covers. It depend on the final result that you want to achieve.

## Second way
The second way is to implement a new fill strategy that constructs item sequences in the right order to meet requirements.

**Comment by umitgoren** [comment]
Thanks. I will prefer the first way.
